OpenRA.Game/Network/OrderManager.cs

```diff
@@ -35,11 +40,44 @@ public sealed class OrderManager : IDisposable
 		public bool AuthenticationFailed = false;
 		public ExternalMod ServerExternalMod = null;
 
+		public int NetTickScale { get { return LobbyInfo.GlobalSettings.UseNewNetcode ? Game.NewNetcodeNetTickScale : Game.DefaultNetTickScale; } }
+		public bool IsNetTick { get { return LocalFrameNumber % NetTickScale == 0; } }
 		public int NetFrameNumber { get; private set; }
 		public int LocalFrameNumber;
-		public int FramesAhead = 0;
 
+		public int SyncFrameScale = 1; // TODO make this based on actual time, I would suggest once per second
+
+		public int LastSlowDownRequestTick;
+		public bool ShouldUseCatchUp;
+		public int OrderLatency; // Set during lobby by a "SyncInfo" packet, see UnitOrders
+		public int NextOrderFrame;
+		public int CatchUpAmount { get; private set; }
+		public int SuggestedTimestep
+		{
+			get
+			{
+				if (World == null)
+					return Game.Timestep;
+
+				if (!ShouldUseCatchUp)
+					return World.Timestep;
+
+				if (IsStalling || World.IsLoadingGameSave)
+					return 1;
+
+				return TargetTimestep;
+			}
+		}
+
+		// int suggestedCatchupTimestep;
+		public volatile bool IsStalling;
 		public long LastTickTime = Game.RunTime;
+		public long RealTickTime = Game.RunTime;
```

Comment:
I found these names quite confusing. Can we please use `TickStartRuntime` and `LastTickStartRuntime` to match the existing naming convention?